sinks/kafka/driver.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/optiopay/kafka"
+	"github.com/optiopay/kafka/proto"
+	"k8s.io/heapster/extpoints"
+	sink_api "k8s.io/heapster/sinks/api"
+	kube_api "k8s.io/kubernetes/pkg/api"
+	"net/url"
+	"time"
+)
+
+const (
+	partition                int32         = 0
+	brokerClientID           string        = "kafka-sink"
+	brokerDialTimeout        time.Duration = 10 * time.Second
+	brokerDialRetryLimit     int           = 10
+	brokerDialRetryWait      time.Duration = 500 * time.Millisecond
+	brokerAllowTopicCreation bool          = true
+	brokerLeaderRetryLimit   int           = 10
+	brokerLeaderRetryWait    time.Duration = 500 * time.Millisecond
+)
+
+type kafkaSink struct {
+	producer        kafka.Producer
+	timeSeriesTopic string
+	eventsTopic     string
+	sinkBrokerHosts []string
+}
+
+// START: ExternalSink interface implementations
+
+func (self *kafkaSink) Register(mds []sink_api.MetricDescriptor) error {
+	return nil
+}
+
+func (self *kafkaSink) Unregister(mds []sink_api.MetricDescriptor) error {
+	return nil
+}
+
+func (self *kafkaSink) StoreTimeseries(timeseries []sink_api.Timeseries) error {
+	if timeseries == nil || len(timeseries) <= 0 {
+		return nil
+	}
+	for index := range timeseries {
+		err := self.produceKafkaMessage(timeseries[index], self.timeSeriesTopic)
+		if err != nil {
+			glog.Errorf("failed to produce Kafka messages : %s", err)
+			return fmt.Errorf("failed to produce Kafka messages: %s", err)
+		}
+	}
+	return nil
+}
+
+func (self *kafkaSink) StoreEvents(events []kube_api.Event) error {
+	if events == nil || len(events) <= 0 {
+		return nil
+	}
+	for index := range events {
+		err := self.produceKafkaMessage(events[index], self.eventsTopic)
+		if err != nil {
+			glog.Errorf("failed to produce Kafka messages: %s", err)
+			return fmt.Errorf("failed to produce Kafka messages: %s", err)
```

Comment:
I see logs and errors duplicated everywhere. Is this necessary? If it is, please deduplicate the strings.